### PR TITLE
enables frontend scaleout

### PR DESCRIPTION
**Pull request by jeremymv2**
Enable a deployment pattern of horizontally scaling out front-end API nodes via the `install.sh` script.

Signed-off-by: Jeremy J. Miller <jm@chef.io>

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -12,9 +12,12 @@ cat NOTICE
 echo
 
 license="${HAB_LICENSE:-}"
+export response
```
Comment (smacfarlane) [inline install.sh:15]: Nit: It would be safer to use `declare` here. `export` will make the `response` variable available to processes spawned by this script, which could cause undesired and unexpected behavior.  We're protected here since we use `sudo` to call the remaining scripts and we don't ever call the `s3migrate.sh` script from here, where `response` is also used. 

I don't feel this needs to change, but given the amount of shell we have in this project felt it was still worth mentioning.

File: scripts/provision.sh [hunk 2]
```diff
@@ -35,19 +51,23 @@ EOT
...
-      sleep 2
-    done
-
-    PGUSER='hab'
-    PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile)
-  else
-    PGUSER=${RDS_USER:-hab}
-    PGPASSWORD=${RDS_PASSWORD:-hab}
-  fi
+    if [ "${RDS_ENABLED:-false}" = "false" ]; then
```
Comment (smacfarlane) [inline scripts/provision.sh:54]: Should this be `==`?

File: scripts/provision.sh [hunk 3]
```diff
@@ -65,15 +85,18 @@ configure() {
 
   # don't write out the builder-minio user.toml if using S3 or Artifactory directly
   if [ "${S3_ENABLED:-false}" = "false" ] && [ "${ARTIFACTORY_ENABLED:-false}" = "false" ]; then
```
Comment (smacfarlane) [inline scripts/provision.sh:87]: It's not part of this pr, but should these also be `==`?

File: scripts/provision.sh [hunk 4]
```diff
@@ -208,7 +236,7 @@ generate_bldr_keys() {
 }
 
 upload_ssl_certificate() {
-  if [ ${APP_SSL_ENABLED} = true ]; then
+  if [ "${APP_SSL_ENABLED}" = "true" ]; then
```
Comment (smacfarlane) [inline scripts/provision.sh:239]: Should this also be `==`?

File: scripts/provision.sh [hunk 5]
```diff
@@ -220,7 +248,25 @@ upload_ssl_certificate() {
...
+  echo "Starting Builder Frontend Services"
+  start_memcached
+  start_api
+  start_api_proxy
+}
+
 start_builder() {
+  echo
+  echo "Starting Builder Services"
   if [ "${RDS_ENABLED:-false}" = "false" ]; then
```
Comment (smacfarlane) [inline scripts/provision.sh:270]: It's not part of this pr, but should these also be `==`?

File: scripts/provision.sh [hunk 6]
```diff
@@ -229,25 +275,152 @@ start_builder() {
   if [ "${ARTIFACTORY_ENABLED:-false}" = "false" ] && [ "${S3_ENABLED:-false}" = "false" ]; then
```
Comment (smacfarlane) [inline scripts/provision.sh:275]: It's not part of this pr, but should these also be `==`?

File: scripts/provision.sh (hunk 2 above)
Comment (eeyun) [reply scripts/provision.sh:54]: Nope. As I understand it, the `[]` in bash are a synonym for `test` and for the `test` syntax this is correct.

File: scripts/provision.sh (hunk 3 above)
Comment (eeyun) [reply scripts/provision.sh:87]: Nope. As I understand it, the `[]` in bash are a synonym for `test` and for the `test` syntax this is correct.

File: scripts/provision.sh (hunk 4 above)
Comment (eeyun) [reply scripts/provision.sh:239]: Nope. As I understand it, the `[]` in bash are a synonym for `test` and for the `test` syntax this is correct.

File: scripts/provision.sh (hunk 5 above)
Comment (eeyun) [reply scripts/provision.sh:270]: Nope. As I understand it, the `[]` in bash are a synonym for `test` and for the `test` syntax this is correct.

File: scripts/provision.sh (hunk 6 above)
Comment (eeyun) [reply scripts/provision.sh:275]: Nope. As I understand it, the `[]` in bash are a synonym for `test` and for the `test` syntax this is correct.

File: install.sh (hunk 1 above)
Comment (eeyun) [reply install.sh:15]: Excellent catch! I think fixing this is the right thing.

File: scripts/provision.sh (hunk 2 above)
Comment (smacfarlane) [reply scripts/provision.sh:54]: That makes a ton more sense now.  Thanks for connecting those dots for me 😸

File: on-prem-docs/scaling.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Front-end Scaling
+With any tiered or HA deployment of the builder services you'll likely want to horizontally scale your front-end nodes. The most common deployment pattern for this usecase is a pool of front-end nodes fronted by a load-balancer.
+
+## Deploying New Front-ends
+The on-prem-builder install.sh script now supports scaling front-end nodes as a deployment pattern. It is require that new front-ends be deployed on a separate compute from your initial on-prem deployment. Similarly to Chef Automate's bootstrap pattern the on-prem builder install script can generate a bootstrap bundle which is used to simplify the deployment of new front-ends.
```
Comment (markan) [inline on-prem-docs/scaling.md:5]: Is there guidance on what provisioning requirements we have for FE nodes? I see we have guidance for the system in general, but might be worth calling out minimum suggested system.
